Add a bootupd section (#203)
coreos/fedora-coreos-docs: modules/ROOT/pages/bootloader-updates.adoc

>  .

Reviewer: Stray dot?

= Updating the bootloader

== bootupd

Updating the bootloader is not currently automatic.  The https://github.com/coreos/bootupd/[bootupd]
project is included in Fedora CoreOS and may be used for manual updates.
This is usually only relevant on bare metal scenarios, or virtualized

> hypervisors that support Secure Boot.

Reviewer: Can we expand a bit on why users might want to update, and how they'll know that that situation applies to them?

Inspect the system status:

[source,bash]
----
# bootupctl status
Component EFI
  Installed: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
  Update: At latest version
----

If an update is available, use `bootupctl update` to apply it; the
change will take effect for the next reboot.

[source,bash]
----
# bootupctl update
...
Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
----

=== Using images that predate bootupd

Older CoreOS images that predate the existence of bootupd need

> an explicit "adoption" phase:

Reviewer: How will users know if they need to do this?

Author: It's visible in `bootupctl status`.  I know it could be improved, but this PR is more trying to document the current state.

= Updating the bootloader

== bootupd

Updating the bootloader is not currently automatic.  The https://github.com/coreos/bootupd/[bootupd]
project is included in Fedora CoreOS and may be used for manual updates.

This is usually only relevant on bare metal scenarios, or virtualized
hypervisors that support Secure Boot.  An example reason to update the

> bootloader is for https://eclypsium.com/2020/07/29/theres-a-hole-in-the-boot/[the BootHole vulnerablity]/

Reviewer: `vulnerability`

Trailing slash should be trailing period.

Reviewer: `vulnerability` is still misspelled.

= Updating the bootloader

> == bootupd

Reviewer 2: Should we mention that this is just for EFI for now?

Author: Done.

Updating the bootloader is not currently automatic.  The https://github.com/coreos/bootupd/[bootupd]
project is included in Fedora CoreOS and may be used for manual updates.

This is usually only relevant on bare metal scenarios, or virtualized
hypervisors that support Secure Boot.  An example reason to update the

> bootloader is for https://eclypsium.com/2020/07/29/theres-a-hole-in-the-boot/[the BootHole vulnerablity].

Reviewer 3: ```suggestion
bootloader is for https://eclypsium.com/2020/07/29/theres-a-hole-in-the-boot/[the BootHole vulnerability].
```

Inspect the system status:

[source,bash]
----
# bootupctl status
Component EFI
  Installed: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
  Update: At latest version
#
----

If an update is available, use `bootupctl update` to apply it; the
change will take effect for the next reboot.

[source,bash]
----
# bootupctl update
...
Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
#
----

> .Example systemd unit to automate bootupd updates

Reviewer 2: Should we have a `Note:` somewhere around here to say that one may not actually want to update the bootloader automatically on every host update?

Author: Not sure what to say; either you update it automatically or you don't, right?

This is usually only relevant on bare metal scenarios, or virtualized
hypervisors that support Secure Boot.  An example reason to update the
bootloader is for https://eclypsium.com/2020/07/29/theres-a-hole-in-the-boot/[the BootHole vulnerablity].

Inspect the system status:

[source,bash]
----
# bootupctl status
Component EFI
  Installed: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
  Update: At latest version
#
----

If an update is available, use `bootupctl update` to apply it; the
change will take effect for the next reboot.

[source,bash]
----
# bootupctl update
...
Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
#
----

.Example systemd unit to automate bootupd updates
[source,yaml]
----
variant: fcos
version: 1.1.0
systemd:
  units:
    - name: custom-bootupd-auto.service
      enabled: true
      contents: |
        [Unit]
        Description=Bootupd automatic update

        [Service]
        ExecStart=/usr/bin/bootupctl update
        RemainAfterExit=yes

        [Install]
        WantedBy=multi-user.target
----

=== Using images that predate bootupd

Older CoreOS images that predate the existence of bootupd need

> an explicit "adoption" phase.  You can see this by looking at
> `bootupctl status` and if it says the component is "Adoptable".

Reviewer: ```suggestion
an explicit "adoption" phase.  If `bootupctl status` says the component
is `Adoptable`, perform the adoption with `bootupctl adopt-and-update`.
```

[source,bash]
----
# bootupctl adopt-and-update
...
Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
#
----

=== Future versions may default to automatic updates

It is possible that future Fedora CoreOS versions may default
to automating bootloader updates similar to the above.
If you choose to add a systemd unit per above (or manually
`ssh` to a node for updates), note that concurrent
`bootupctl update` invocations will not explicitly conflict.
The only side effect is that without explicit ordering,
one invocation may see and perform an update, and

> the second may detect no update available.

Reviewer: Is this advice actionable?  If we recommend users add a custom systemd unit, we'll need to continue supporting that unit in the future.  If it's harmless to run the custom unit alongside a hypothetical standard one, I think we can just drop everything but the first sentence of this section.
